File: src/main/java/land/leets/domain/comment/presentation/dto/CommentResponse.java
package land.leets.domain.comment.presentation.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import land.leets.domain.admin.presentation.dto.AdminDetailsResponse;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
public class CommentResponse {

    @NotBlank
    private String content;

    @NotBlank
    private LocalDateTime createdAt;

    @NotBlank
    private LocalDateTime updatedAt;

    @NotNull
    private AdminDetailsResponse admin;
}

File: src/main/kotlin/land/leets/domain/comment/presentation/dto/CommentResponse.kt
package land.leets.domain.comment.presentation.dto

import land.leets.domain.admin.presentation.dto.AdminDetailsResponse
import land.leets.domain.comment.domain.Comment
import java.time.LocalDateTime

data class CommentResponse(
    val content: String,

    val createdAt: LocalDateTime?,

    val updatedAt: LocalDateTime?,

    val admin: AdminDetailsResponse
) {
    companion object {
        fun from(comment: Comment): CommentResponse {
            return CommentResponse(
                content = comment.content,
                createdAt = comment.createdAt,
                updatedAt = comment.updatedAt,
                admin = AdminDetailsResponse.from(comment.admin)
            )
        }
    }
}
